pingcap/tiflash
# Feature/support cast enum to int/real/decimal

### What problem does this PR solve?

Issue Number: #3034 

Problem Summary: sum(enum_val) is not supported to be pushed down)

### What is changed and how it works?

Proposal: support pushdown of sum(enum_val)

What's Changed: add type check and conversion for enum

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch:

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
uncomment tidb's pushdown check of enum
CREATE TABLE shirts (
    name VARCHAR(40),
    size ENUM('x-small', 'small', 'medium', 'large', 'x-large')
);
INSERT INTO shirts (name, size) VALUES ('dress shirt','large'), ('t-shirt','medium'),
  ('polo shirt','small');
desc select sum(cast(size as decimal)) from shirts;
desc select sum(cast(size as signed)) from shirts;
desc select sum(cast(size as unsigned)) from shirts;
desc select sum(size) from shirts;
desc select sum(size + 0) from shirts;
select sum(cast(size as decimal)) from shirts;
select sum(cast(size as signed)) from shirts;
select sum(cast(size as unsigned)) from shirts;
select sum(size) from shirts;
select sum(size + 0) from shirts;

- No code

Side effects



### Release note 

```release-note
support sum(enum_val) in TiDB pushdown query & Clickhouse query
```
